Message Processor Instructions (#3013)
wso2/product-ei: gh-docs/Admin_Services.md

# How to create admin-services

> Admin Services generally follow the following steps. Here, the Message Processor has been considered as an example

Reviewer: ```suggestion
When creating Admin Services you would generally follow the same set of steps. In this instance, the Message Processor is used as an example.
```

### Creation Flow

1. Define your Admin Service function
2. Generate the stub using WSDL

> 3. Define function in MessageProcessorAdminServiceClient to call Admin Service function

Reviewer: ```suggestion
3. Define the function in `MessageProcessorAdminServiceClient` to call the Admin Service function
```

Shall we follow a convention of having all class names, file names, method names, etc. between single backticks? This is to highlight this to readers.

Author: I will make that change.

> 4. Call the function from the Message Processor UI jsp

Reviewer: ```suggestion
4. Call the function from the Message Processor UI JSP
```

### 1. Defining your Admin Service

> Message Processor Admin Services are defined in the 'MessageProcessorAdminService.java' class in the carbon-mediation repo.

Reviewer: Shall we add the link to the carbon-mediation repo?

Author: I am making the necessary changes in a different pull request. I will link the PR here as well once I am done.

Go to `<repo-home>/components/mediation-admin/org.wso2.carbon.message.processor/` and navigate to the 'service' directory
to find the 'MessageProcessorAdminService.java' class.

> Once you define your function, you will have to generate the MessageProcessorAdminServiceStub file. 

Reviewer: ```suggestion
Once you define your function, generate the `MessageProcessorAdminServiceStub` file. 
```
